ntbs-service/Services/AlertService.cs

```diff
@@ -118,22 +118,39 @@ public async Task DismissAllOpenAlertsForNotification(int notificationId)
 
             foreach (var alert in alerts)
             {
+                // notification may have a duplicate alert to be dismissed on multiple other records
                 if (alert is DataQualityPotentialDuplicateAlert duplicateAlert)
                 {
-                    var duplicate =
-                        await _notificationRepository.GetNotificationForAlertCreationAsync(duplicateAlert.DuplicateId);
+                    await DismissDuplicationAlertsOnDuplicateRecord(notificationId, duplicateAlert.DuplicateId);
+                }
+                
+                CloseAlert(alert);
+            }
 
-                    if (duplicate.NotificationStatus.IsOpen())
+            await _alertRepository.SaveAlertChangesAsync();
+        }
+
+        private async Task DismissDuplicationAlertsOnDuplicateRecord(int notificationId, int duplicateAlertId)
+        {
+            var duplicateRecordAlerts = await _alertRepository.GetAllOpenAlertsByNotificationId(duplicateAlertId);
+            if (duplicateRecordAlerts != null)
+            {
+                foreach (var alert in duplicateRecordAlerts)
+                {
+                    if (alert is DataQualityPotentialDuplicateAlert duplicateAlert && 
+                        duplicateAlert.DuplicateId == notificationId)
+                        // Only want to dismiss duplicate alerts where the duplicateId is equal to the original notificationId being closed
                     {
-                        continue;
+                        CloseAlert(duplicateAlert);
                     }
                 }
-
-                alert.ClosureDate = DateTime.Now;
-                alert.AlertStatus = AlertStatus.Closed;
             }
+        }
 
-            await _alertRepository.SaveAlertChangesAsync();
+        private static void CloseAlert(Alert alert)
+        {
+            alert.ClosureDate = DateTime.Now;
+            alert.AlertStatus = AlertStatus.Closed;
```

Comment:
There is a `DismissAlertAsync()` method already in this class - is there a reason we don't use it?
It looks like we actually have a few methods in this class which do some form of dismissing. They could all use a method like this to add the date and set status, so this method would be something like `SetClosedAndAddClosureDate()`

Reply:
I didn't use the pre-existing DismissAlertAsync method because it takes in an alertId and then makes a call to the database to get and close it, whereas we already have a list of all alerts - I will refactor the CloseAlert method and make use of it in a few other places